.github/workflows/ubuntu.yml

```diff
@@ -83,3 +90,92 @@ jobs:
         run: |
           eval $(opam env)
           smoke-test-kit/run-smoke-test.sh
+
+      - name: 'Upload smoke test kit'
+        uses: actions/upload-artifact@v2
+        with:
+          name: 'Smoke Test Kit ${{matrix.variant}}'
+          path: smoke-test-kit
+          retention-days: 5
+
+  Ubuntu_snap:
+    name: Snap package
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Git checkout
+        uses: actions/checkout@v2
+
+      - name: Set PLATFORM
+        if: ${{ github.event.inputs.platform != '' }}
+        run: echo "PLATFORM=${{ github.event.inputs.platform }}" >> $GITHUB_ENV
+
+      - name: Set SNAP_PICK
+        if: ${{ github.event.inputs.snap_pick != '' }}
+        run: echo "SNAP_PICK=${{ github.event.inputs.snap_pick }}" >> $GITHUB_ENV
+
+      - name: Generate snapcraft file
+        run: linux/create_snapcraft_yaml.sh -packages=$SNAP_PICK $PLATFORM
+
+      - name: Print snapcraft file
+        run: cat snap/snapcraft.yaml
+
+      - name: Run snapcraft
+        uses: snapcore/action-build@v1.0.9
+        id: build
+
+      - name: Save Artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: 'Snap package'
+          path: ${{ steps.build.outputs.snap }}
+
+      - name: Upload Artifact to the Snap Store
+        if: ${{ github.event.inputs.upload }}
+        uses: snapcore/action-publish@v1
+        with:
+          store_login: ${{ secrets.STORE_LOGIN }}
+          snap: ${{ steps.build.outputs.snap }}
+          release: edge
+
+  # We run this one always because the Ubuntu_platform job
```

Comment:
@gares: I thought about disabling the dev job except for scheduled builds. What is your opinion on this?

Reply:
works for me, but then schedled build would likely not smoke test the snap.
I guess the best is to do both: always run the smoke step, and disable dev if not cron